modules/io/io/io_factory.cc

```diff
@@ -41,12 +41,10 @@ void IOFactory::Finalize() { LocalIOAdaptor::Finalize(); }
  */
 std::unique_ptr<IIOAdaptor> IOFactory::CreateIOAdaptor(
     const std::string& location, Client*) {
-  std::error_code err;
-  network::uri instance(location, err);
-  std::string scheme;
-  if (!err) {
-    scheme = instance.scheme().to_string();
-  } else {
+  size_t pos = location.find_first_of(':');
+  std::string scheme = location.substr(0, pos);
```

Comment:
Why not `network::uri`? The target of `network:uri` is to avoid ad-hoc parsing routines.

Reply:
Because it fails to parse uri containing column names and types

Reply:
If uri lib cannot parse out uri, then it's the problem of out uri design. We can discuss how to design the oss uri.

Reply:
https://code.aone.alibaba-inc.com/GraphScope/libgrape-alibaba/blob/master/grape/internal/io/tunnel_io_adaptor.cc#L703
This may help.